File: package/main/src/String/toFullWidth.ts
/**
 * Convert half-width alphanumeric characters to full-width characters.
 * Inverse of `toHalfWidth`; only digits, uppercase and lowercase ASCII
 * letters are converted.
 * @param {string} string_ - String to convert
 * @returns {string} - Converted string
 * @example
 * toFullWidth("Abc123"); // "Ａｂｃ１２３"
 */
export const toFullWidth = (string_: string): string =>
  string_.replaceAll(/[\dA-Za-z]/g, (s) => {
    const code = s.codePointAt(0);
    // biome-ignore lint/style/noNonNullAssertion: code is not null
    return String.fromCodePoint(code! + 0xfe_e0);
  });

File: package/umt_python/src/string/to_full_width.py
import re


def to_full_width(string_: str) -> str:
    """
    Convert half-width alphanumeric characters to full-width characters.
    Inverse of :func:`to_half_width`; only digits, uppercase and lowercase
    ASCII letters are converted.

    Args:
        string_: String to convert.

    Returns:
        Converted string.

    Example:
        >>> to_full_width("Abc123")
        'Ａｂｃ１２３'
        >>> to_full_width("a-b!")
        'ａ-ｂ!'
        >>> to_full_width("")
        ''
    """
    return re.sub(
        r"[0-9A-Za-z]",
        lambda match: chr(ord(match.group(0)) + 0xFEE0),
        string_,
    )
